src/main/java/com/epam/ta/reportportal/ws/controller/LaunchAsyncController.java

```diff
@@ -0,0 +1,124 @@
+import com.epam.ta.reportportal.core.launch.util.LaunchLinkGenerator;
+import com.epam.ta.reportportal.ws.model.BulkRQ;
+import com.epam.ta.reportportal.ws.model.FinishExecutionRQ;
+import com.epam.ta.reportportal.ws.model.OperationCompletionRS;
+import com.epam.ta.reportportal.ws.model.launch.FinishLaunchRS;
+import com.epam.ta.reportportal.ws.model.launch.StartLaunchRQ;
+import com.epam.ta.reportportal.ws.model.launch.StartLaunchRS;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.core.annotation.AuthenticationPrincipal;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.servlet.http.HttpServletRequest;
+import java.util.List;
+
+import static com.epam.ta.reportportal.auth.permissions.Permissions.ALLOWED_TO_REPORT;
+import static com.epam.ta.reportportal.commons.EntityUtils.normalizeId;
+import static com.epam.ta.reportportal.util.ProjectExtractor.extractProjectDetails;
+import static org.springframework.http.HttpStatus.CREATED;
+import static org.springframework.http.HttpStatus.OK;
+
+/**
+ * Controller implementation for async reporting client API for
+ * {@link com.epam.ta.reportportal.entity.launch.Launch} entity
+ *
+ * @author Konstantin Antipin
+ */
+@RestController
+@RequestMapping("/{projectName}/async/launch")
+public class LaunchAsyncController {
+
+	private final StartLaunchHandler createLaunchMessageHandler;
+	private final FinishLaunchHandler finishLaunchMessageHandler;
+
+	@Autowired
+	public LaunchAsyncController(@Qualifier("startLaunchHandlerAsync") StartLaunchHandler createLaunchMessageHandler,
+                                 @Qualifier("finishLaunchHandlerAsync") FinishLaunchHandler finishLaunchMessageHandler) {
+		this.createLaunchMessageHandler = createLaunchMessageHandler;
+		this.finishLaunchMessageHandler = finishLaunchMessageHandler;
+	}
+
+	@Transactional
+	@PostMapping
+	@PreAuthorize(ALLOWED_TO_REPORT)
+	@ResponseStatus(CREATED)
+	@ApiOperation("Starts launch for specified project")
+	public StartLaunchRS startLaunch(@PathVariable String projectName,
+									 @ApiParam(value = "Start launch request body", required = true) @RequestBody @Validated StartLaunchRQ startLaunchRQ,
```

Comment:
Can you configure please our code formatting
https://github.com/reportportal/reportportal/blob/master/idea_formatting_profile.xml